packages/flutter/lib/src/material/dialog.dart

```diff
@@ -252,10 +265,14 @@ class AlertDialog extends StatelessWidget {
 
     return new Dialog(
       child: new IntrinsicWidth(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+        child: new Semantics(
```

Comment:
Something that just crossed my mind while reading this: Can you make sure that this semantics node is not focusable with TalkBack on Android and VoiceOver? My suspicion is that you might actually be able to focus on that node because on both iOS and Android we say something is focusable when it has a value:
https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L877

https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm#L167

Reply:
You are right, this is focusing the page itself... I can fix this in the engine by excluding route flags from focusability.

Reply:
Fixed in https://github.com/flutter/engine/pull/4901/commits/6fcb204f80ab35c65429b399e8cc1cc4604eb382